### PR TITLE
MAINT: sparse: update tests to switch to format and toarray from get_format() and .A, .H

### PR DESCRIPTION
**Pull request by dschult**
The changes `spA.get_format()` -> `spA.format` and `spA.A` -> `spA.toarray()` bring our sparse tests closer to the array syntax, replacing some of these matrix based idioms in the tests. It's a small step toward leaving the `np.matrix` api, but its a step (and doesn't affect users -- this is only the tests).

This concern was [raised in the 1d coo array](https://github.com/scipy/scipy/pull/18530#discussion_r1314282239) PR #18530 but is really separate from that PR. So for simplicity it's on its own here.

### CODE REVIEW

File: scipy/sparse/tests/test_base.py
Comment (j-bowhay) [inline scipy/sparse/tests/test_base.py:4670]: Here and else where: the addition of `toarray()` means the array is no longer correctly aligned

Comment (dschult) [reply scipy/sparse/tests/test_base.py:4670]: Thanks! They're realigned.

**Review by tylerjereddy — COMMENTED** [review]
The diff seems sensible, maybe best for i.e., CJ to double check.

Let me restart the 32-bit CI job, probably unrelated but that's a different one than recently.

```
 scipy/sparse/linalg/_eigen/lobpcg/lobpcg.py:121: in _b_orthonormalize
    if blockVectorBV.shape != blockVectorV.shape:
E   AttributeError: 'NoneType' object has no attribute 'shape'
        B          = <function _makeMatMat.<locals>.<lambda> at 0xd690d100>
        blockVectorBV = None
        blockVectorV = array([[ 0.11057004-0.70986813j],
       [ 0.26274613+0.22422072j],
       [-0.05223355-0.6099342j ],
       [ 0.37218...       [ 0.30558693-0.06317747j],
       [ 0.08693056-0.57411754j],
       [ 0.20404057-0.61137146j]], dtype=complex64)
        verbosityLevel = 0
=========================== short test summary info ============================
FAILED scipy/sparse/linalg/_eigen/lobpcg/tests/test_lobpcg.py::test_hermitian
```

**Review by j-bowhay — APPROVED** [review]

**Review by stefanv — APPROVED** [review]

**Comment by stefanv** [comment]
Test failures seem unrelated.

**Comment by tylerjereddy** [comment]
It has two core dev approvals now, and the CI fails are unrelated.

Also, it has a `1.12.0` milestone and we're two weeks from branching so for this and other fairly clear cases I'll probably start merging a bit faster if there's no clear problem, etc.

**Comment by tylerjereddy** [comment]
thanks!

**Pull request merged**
